crates/matrix-sdk-indexeddb/src/crypto_store/mod.rs

```diff
@@ -746,50 +783,85 @@ impl_crypto_store! {
             .get_all()?
             .await?
             .iter()
-            .filter_map(|i| self.serializer.deserialize_value(i).ok())
-            .filter_map(|p| InboundGroupSession::from_pickle(p).ok())
+            .filter_map(|v| self.deserialize_inbound_group_session(v).ok())
             .collect())
     }
 
     async fn inbound_group_session_counts(&self) -> Result<RoomKeyCounts> {
-        let all = self.get_inbound_group_sessions().await?;
-        let backed_up = all.iter().filter(|s| s.backed_up()).count();
-
-        Ok(RoomKeyCounts { total: all.len(), backed_up })
+        let tx = self
+            .inner
+            .transaction_on_one_with_mode(
+                keys::INBOUND_GROUP_SESSIONS,
+                IdbTransactionMode::Readonly,
+            )?;
+        let store = tx.object_store(keys::INBOUND_GROUP_SESSIONS)?;
+        let all = store.count()?.await? as usize;
+        let not_backed_up = store.index(keys::INBOUND_GROUP_SESSIONS_BACKUP_INDEX)?.count()?.await? as usize;
+        tx.await.into_result()?;
+        Ok(RoomKeyCounts { total: all, backed_up: all-not_backed_up })
```

Comment:
I suppose that's in the macro invoke, but the spacing is wrong around `all-not_backed_up`.